File: python/silapay_connector/silapay_connector/connector.py
from .providers.adyen import AdyenProvider
from .providers.mercadopago import MercadoPagoProvider
from .providers.silapay import SilapayProvider

PROVIDERS = {
    "silapay": SilapayProvider,
    "mercadopago": MercadoPagoProvider,
    "adyen": AdyenProvider,
}


class PaymentConnector:
    def __init__(self, provider="silapay", **options):
        provider_class = PROVIDERS.get(provider)

        if provider_class is None:
            raise ValueError(
                f'Provider "{provider}" nao suportado. Providers disponiveis: silapay, mercadopago, adyen'
            )

        self.provider_name = provider
        self.provider = provider_class(**options)

    def pix(self, dados):
        return self.provider.pix(dados)

    def boleto(self, dados):
        return self.provider.boleto(dados)

    def cartao(self, dados):
        return self.provider.cartao(dados)

    def saldo(self):
        return self.provider.saldo()

File: javascript/silapay-connector/connector.js
const SilapayProvider = require("./providers/silapay");
const MercadoPagoProvider = require("./providers/mercadopago");
const AdyenProvider = require("./providers/adyen");

const PROVIDERS = {
    silapay: SilapayProvider,
    mercadopago: MercadoPagoProvider,
    adyen: AdyenProvider,
};

class PaymentConnector {
    constructor({ provider = "silapay", ...options } = {}) {
        const Provider = PROVIDERS[provider];

        if (!Provider) {
            throw new Error(
                `Provider "${provider}" não suportado. Providers disponíveis: silapay, mercadopago, adyen`
            );
        }

        this.providerName = provider;
        this.provider = new Provider(options);
    }

    async pix(dados) {
        return this.provider.pix(dados);
    }

    async boleto(dados) {
        return this.provider.boleto(dados);
    }

    async cartao(dados) {
        return this.provider.cartao(dados);
    }

    async saldo() {
        return this.provider.saldo();
    }
}

module.exports = PaymentConnector;
